File: src/two_sum_1/solution_hash_table.ts
function twoSum(nums: number[], target: number): number[] {
  const lookupTable: { [key: number]: number } = {};

  for (let idx = 0; idx < nums.length; idx++) {
    const num = nums[idx];
    const find = target - num;

    let idx_find = lookupTable[find];
    if (idx_find !== undefined) {
      return [idx_find, idx];
    }

    lookupTable[num] = idx;
  }

  return [-1, -1];
}

export function test() {
  console.log(twoSum([3, 2, 4], 6));
}

File: src/two_sum_1/solution_hash_table.go
package two_sum_1

import "fmt"

func twoSumHashTable(nums []int, target int) []int {
	lookupTable := make(map[int]int)

	for idx, num := range nums {
		find := target - num

		if idxFind, exists := lookupTable[find]; exists {
			return []int{idxFind, idx}
		}

		lookupTable[num] = idx
	}

	return []int{-1, 1}
}

func TestHashTable() {
	fmt.Println(twoSumHashTable([]int{3, 2, 4}, 6))
}
